lib/Contracts/IMailManager.php

```diff
@@ -128,6 +128,17 @@ public function markFolderAsRead(Account $account, string $folderId): void;
 	 */
 	public function flagMessage(Account $account, string $mailbox, int $uid, string $flag, bool $value): void;
 
+	/**
+	 * @param Account $account
+	 * @param string $mailbox
+	 * @param bool $subscribed
+	 *
+	 * @return array
+	 * @throws ClientException
+	 * @throws ServiceException
+	 */
+	public function updateSubscription(Account $account, string $mailbox, bool $subscribed): array;
```

Comment:
I'm sorry for the confusion. But here the `bool` return type is preferred. We should only change the response data structure in the controller.